### PR TITLE
[fixed-for-master] Survey custom fields in petition UI

### PR DESCRIPTION
**Pull request by agh1**
Petitions are surveys but they don't expose the petition custom fields in the UI.

### CODE REVIEW

File: CRM/Campaign/Form/Petition.php [hunk 1]
```diff
@@ -126,6 +126,10 @@ public function preProcess() {
    *   array of default values
    */
   public function setDefaultValues() {
+    if ($this->_cdType) {
+      return CRM_Custom_Form_CustomData::setDefaultValues($this);
```
Comment (eileenmcnaughton) [inline CRM/Campaign/Form/Petition.php]: Didn't we actually get rid of this from all the other forms? In 4.7 we refactored away the concept of custom form as a distributed form

Comment (eileenmcnaughton) [inline CRM/Campaign/Form/Petition.php]: yep - most instances of _cdType are gone from 4.7

Comment (agh1) [inline CRM/Campaign/Form/Petition.php]: Yeah this is a pr against 4.6.  It might well need a different set of changes for 4.7.

Comment (eileenmcnaughton) [inline CRM/Campaign/Form/Petition.php]: Right & we would want to block it from merging to 4.7 *somehow*

**Comment by colemanw** [comment]
I'm suggesting we do not merge this until 4.6 is handed off to the LTS team.

**Comment by eileenmcnaughton** [comment]
That makes sense in terms of not wanting to merge it up - it would also mean LTS rules are that a fix needs to be in 4.7 before we merge - but of course it would be a different fix.

Ideally we would also find someone with capacity to QA this fix against 4.6 after merging.

**Comment by eileenmcnaughton** [comment]
NB - technically speaking it's an enhancement rather than a fix - I would be willing to accept it for an LTS release because I perceive it as low risk

**Comment by eileenmcnaughton** [comment]
I have changed the title of this PR to indicate that it is blocked by there not being a fix in master at the moment. When that is resolved please re-edit the title - either to remove it or perhaps we should use a new convention like [issues-resolved] or [look-at-me] to draw people's attention to the fact that issues on it have been resolved & it needs re-assessing?

**Comment by agh1** [comment]
Just submitted #7883 to handle the master branch.

**Pull request merged**